api/models/author.py

```diff
@@ -60,15 +60,16 @@ def add_pending(self, friend):
         if not self.pending.filter(id=friend.id).exists():
             self.pending.add(friend)
             self.save()
-
         self.follow(friend)
 
     def add_friend(self, friend):
         """Create a friend from an author model"""
-        friend = self._get_cached_author(friend)
-        if not self.friends.filter(id=friend.id).exists():
-            self.friends.add(friend)
+        friend_cached = self._get_cached_author(friend)
+        if not self.friends.filter(id=friend_cached.id).exists():
+            self.friends.add(friend_cached)
             self.save()
+            friend.add_friend(self)
```

Comment:
I think using the ModifyRelationsMixin is still the best approach. Alternatively, we can also use signals to check every time a friends/followers attribute has been modified and then update the corresponding local author.